databricks/koalas/frame.py

```diff
@@ -5243,12 +5213,13 @@ def replace(
         3    3    8  d
         4    4    9  e
 
-        Notes
-        -----
-        One difference between this implementation and pandas is that it is necessary
-        to specify the column name when you are passing dictionary in `to_replace`
-        parameter. Calling `replace` on its index such as `df.replace({0: 10, 1: 100})` will
-        throw an error. Instead specify column-name like `df.replace({'A': {0: 10, 1: 100}})`.
+        >>> df.replace({0: 10, 1: 100})
```

Comment:
Hm.. just realised that it works different from pandas assuming from https://github.com/databricks/koalas/pull/495#issuecomment-505129634.

The key there means a column name but here seems a value.

Reply:
good idea. I'll fix it like that :)

Reply:
@HyukjinKwon 
https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.replace.html

According to the above document, I found that the `key` could be a `value` or a `column`. It also works the same for Koalas. Should I change the documentation of doctest like below?


```python
>>> df = ks.DataFrame({"name": ['Ironman', 'Captain America', 'Thor', 'Hulk'],
        ...                    "weapon": ['Mark-45', 'Shield', 'Mjolnir', 'Smash']},
        ...                   columns=['name', 'weapon'])
        >>> df
                      name   weapon
        0          Ironman  Mark-45
        1  Captain America   Shield
        2             Thor  Mjolnir
        3             Hulk    Smash

        #  For a DataFrame a dict can specify that different values should be replaced in different column, 
        #  To use a dict in this way the value parameter should be None.

        >>> df.replace({'Mjolnir': 'Stormbuster'})
              name       weapon
        0   Rescue      Mark-45
        1  Hawkeye       Shield
        2     Thor  Stormbuster
        3     Hulk        Smash
       
        # For a DataFrame a dict can specify that different values should be replaced in different columns. 
        # The value parameter should not be None in this case.

        >>> df.replace({'weapon': 'Mjolnir'}, 'Stormbuster')
              name       weapon
        0   Rescue      Mark-45
        1  Hawkeye       Shield
        2     Thor  Stormbuster
        3     Hulk        Smash

        # For a DataFrame nested dictionaries. The value parameter should be None to use a nested dict in this way.

        >>> df.replace({'weapon': {'Mjolnir': 'Stormbuster'}})
              name       weapon
        0   Rescue      Mark-45
        1  Hawkeye       Shield
        2     Thor  Stormbuster
        3     Hulk        Smash
```

Reply:
Hmm, the link you shared https://github.com/databricks/koalas/pull/495#issuecomment-505129634 shows that it may work differently inside Spark. Let me investigate a little bit more.

Reply:
I think it will be possible to create `subset` inside` replace` function 
according to the form of input dictionary and pass it to Spark.

Reply:
https://github.com/databricks/koalas/pull/495#issuecomment-505142114
This link shows that, he was forced to use a `subset` for index integrity. However, in the code now, a `subset` is created inside (Line 5241) and passed to spark's replace function. Therefore, it seems that there is no problem because the index integrity is guaranteed.


```python
>>> kdf
   A  B  C
0  0  5  a
1  1  6  b
2  2  7  c
3  3  8  d
4  4  9  e
>>> pdf = kdf.to_pandas()
>>> pdf
   A  B  C
0  0  5  a
1  1  6  b
2  2  7  c
3  3  8  d
4  4  9  e
>>> pdf.replace([0, 1, 2, 3], [4, 3, 2, 1])
   A  B  C
0  4  5  a
1  3  6  b
2  2  7  c
3  1  8  d
4  4  9  e
>>> replaced_kdf = kdf.replace([0, 1, 2, 3], [4, 3, 2, 1])
>>> replaced_kdf
   A  B  C
0  4  5  a
1  3  6  b
2  2  7  c
3  1  8  d
4  4  9  e
>>> replaced_kdf._internal._sdf.show()
+-----------------+---+---+---+-----------------+
|__index_level_0__|  A|  B|  C|__natural_order__|
+-----------------+---+---+---+-----------------+
|                0|  4|  5|  a|       8589934592|
|                1|  3|  6|  b|      25769803776|
|                2|  2|  7|  c|      34359738368|
|                3|  1|  8|  d|      51539607552|
|                4|  4|  9|  e|      60129542144|
+-----------------+---+---+---+-----------------+

```